exploration/selection-matching-options.md

```diff
@@ -0,0 +1,547 @@
+when *   0   one {Congratulations, you earned {$coins} coin}
+when *   0   *   {Congratulations, you earned {$coins} coins}
+when *   one one {You have {$days} days to purchase {$items} item to earn {$coins} coin}
+when *   one *   {You have {$days} days to purchase {$items} item to earn {$coins} coins}
+when *   *   one {You have {$days} day to purchase {$items} items to earn {$coins} coin}
+when *   *   *   {You have {$days} days to purchase {$items} items to earn {$coins} coins}
+```
+
+## First-Match
+
+First-Match selection evaluates the list of _keys_ row-by-row and selects the first _variant_ that matches the list of _selectors_ as the _pattern_.
+
+In the example message, the _variants_ are in a canonical order, so first-match produces the same order as best-match does.
+
+**Pros**
++ Allows developers to control the order of selection.
++ Allows translators to tailor the order of selection.
++ Can visually inspect match order.
++ May be more efficient when perfoming match (??)
+
+**Cons**
+- Requires developers to specify _variants_ in the correct order.
+- Requires translators to tailor the order of _variants_ if this is different from the source.
+- Requires all translation tooling and runtime processing to preserve the order of the _variants_
+- Translation tools that "explode" the selection matrix to support languages with additional needs (such as the way that the `pl` locale requires keywords `few` and `many` for plurals whereas the `en` locale does not) must order the generated _variants_ correctly.
+
+## Best-Match
+
+Best-Match selection evaluates the full list of _keys_ and selects the _variant_ that, according to the matching strategy used, "best matches" the list of _selectors_ as the _pattern_. There are three best-match selection strategies in this document:
+
+1. Sorted Matching
+1. Column-First with required default `*`
+1. Column-First without required default `*`
+
+**Pros**
++ Variants can be written in any order and produce a consistent result.
++ Selector developers can write complex matches that produce different quality matches for the same value. For example, `{|1| :plural}` matches both the variant `1` and the variant `one`, but prefers the value `1`. The plural _selector_ does not need to communicate with the other _selectors_ in order to arrive at the best matching pattern.
++ Translators do not need to worry about the order of variants or need to reorder variants (which can be difficult to do when only the translation segment for the pattern is shown or when only a changed or generated _variant_ is exposed to translation.
++ Translation tools do not have to preserve the order of _variants_ and are free to send only the translatable segment (the pattern) for translation.
+
+**Cons**
+- Developers cannot override the order that the _selector_ provides unless this is exposed as a feature of the given _selector_.
+- More complex matching implementation; may be slower?
+
+### Sorted
```

Comment:
Is this section meant as the base on which the following ones build? Or just some of the following? In any case, I was a bit confused by the fact that it's presented as a fourth alternative while the list above mentions only three.